app/modules/general/components/visibility_selector.svelte

```diff
@@ -0,0 +1,146 @@
+<script>
+  import { I18n } from '#user/lib/i18n'
+  import { userGroups } from '#user/user_groups_store'
+  import { uniq, without } from 'underscore'
+  import { getGroupVisibilityKey, isNotGroupVisibilityKey, commonVisibilityKeys } from '#general/lib/visibility'
+  import { onChange } from '#lib/svelte'
+
+  export let visibility = []
+
+  // Needs to be above reactive call to initCheckedGroupKeys
+  $: allGroupsVisibilityKeys = $userGroups.map(getGroupVisibilityKey)
+
+  // Group keys will be added to 'checked' once $userGroups will have been populated
+  let checked = visibility[0] === 'public' ? commonVisibilityKeys : visibility
+  $: onChange($userGroups, initCheckedGroupKeys)
+
+  function initCheckedGroupKeys () {
+    if (visibility.includes('public') || visibility.includes('groups')) checkAllGroups()
+  }
+
+  function updateCheckedGroupsKeys (allGroupsChecked) {
+    if (allGroupsChecked) checkAllGroups()
+    else uncheckAllGroups()
+  }
+
+  function checkAllGroups () {
+    checked = uniq([ ...checked, 'groups', ...allGroupsVisibilityKeys ])
+  }
+
+  function uncheckAllGroups () {
+    checked = without(checked, 'public', 'groups', ...allGroupsVisibilityKeys)
+  }
+
+  function onPublicClick (e) {
+    if (e.target.checked) {
+      checked = commonVisibilityKeys.concat(allGroupsVisibilityKeys)
+    }
+  }
+
+  function onFriendsClick (e) {
+    if (!e.target.checked) {
+      checked = without(checked, 'public', 'friends')
+    }
+  }
+
+  function onGroupsClick (e) {
+    updateCheckedGroupsKeys(e.target.checked)
+  }
+
+  function onSingleGroupClick (e) {
+    if (!e.target.checked) {
+      checked = without(checked, 'public', 'groups', e.target.value)
+    }
+  }
+
+  $: onChange(checked, updateVisibility)
+
+  function updateVisibility () {
+    if (checked.includes('public')) {
+      visibility = [ 'public' ]
+    } else if (checked.includes('groups')) {
+      visibility = checked.filter(isNotGroupVisibilityKey)
+    } else {
+      visibility = checked
+    }
+  }
+</script>
+
+<fieldset>
+  <legend>{I18n('visibility')}</legend>
+
+  <div class="options">
+    <label>
+      <input
+        type="checkbox"
+        value="public"
+        on:click={onPublicClick}
+        bind:group={checked}
+        >
+      {I18n('public')}
+    </label>
+
+    <label
+      class:inferred={visibility.includes('public')}
+      >
+      <input
+        type="checkbox"
+        value="friends"
+        on:click={onFriendsClick}
+        bind:group={checked}
+      >
+      {I18n('friends')}
+    </label>
+
+    <label
+      class:inferred={visibility.includes('public')}
+      >
+      <input
+        type="checkbox"
+        value="groups"
+        on:click={onGroupsClick}
+
+        bind:group={checked}
+      >
+      {I18n('groups')}
```

Comment:
An explanation besides `groups` would allow lost users between all those checkboxes. Maybe a little icon `info` , with a tooltip ?
i would include the terms "all present and future groups" or any kind of sentence that would represent what checking this box means: setting a flag for all groups ; and not only allowlisting the current groups (listed right below) which is what the current behavior could suggest (checking every groups checkbox when checking `groups`)  .

Reply:
addressed in b27b44b